### PR TITLE
feat(B029): Add B029 from flake8-bugbear

### PR DESCRIPTION
**Pull request by carlosmiei**
- Added B029 from flake8-bugbear
image

- I think I followed all the required steps, but let me know if I missed something

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/flake8_bugbear/B029.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""
+Should emit:
+B029 - on lines 7
+"""
+try:
+    x = 1
+except():
+    print("error")
```
Comment (charliermarsh) [inline crates/ruff/resources/test/fixtures/flake8_bugbear/B029.py:8]: Can we replace this with the version from [flake8-bugbear](https://github.com/PyCQA/flake8-bugbear/blob/main/tests/b029.py) itself?

```py
"""
Should emit:
B029 - on lines 8 and 13
"""

try:
    pass
except ():
    pass

try:
    pass
except () as e:
    pass
```

**Comment by charliermarsh** [comment]
Looking good!

File: crates/ruff/resources/test/fixtures/flake8_bugbear/B029.py (hunk 1 above)
Comment (carlosmiei) [reply crates/ruff/resources/test/fixtures/flake8_bugbear/B029.py:8]: @carlosmiei sure!

Comment (carlosmiei) [reply crates/ruff/resources/test/fixtures/flake8_bugbear/B029.py:8]: @charliermarsh sure!

File: crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    ExceptWithEmptyTuple: ~
+  location:
+    row: 7
+    column: 0
+  end_location:
+    row: 8
+    column: 18
+  fix: ~
+  parent: ~
+
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap:15]: Do you have `insta` installed? (`cargo install cargo-insta`) You should be able to run `cargo test --lib`, then `cargo insta review` to review and check in the generated snapshot.

Comment (carlosmiei) [reply crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap:15]: @charliermarsh  yeah it's weird I did run `cargo test --all` and `cargo insta review` but the file was not updated 🤔

Comment (carlosmiei) [reply crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap:15]: image

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap:15]: Oh, you need to add the test to `crates/ruff/src/rules/flake8_bugbear/mod.rs` (hopefully the pattern is clear once you open up that file).

Comment (carlosmiei) [reply crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap:15]: @charliermarsh nvm 🤦 I accidentally deleted it upon resolving the merge conflict here image

will restore it asap

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B029_B029.py.snap:25]: Happens to the best of us :D

**Comment by charliermarsh** [comment]
Thanks!

**Pull request merged**